### ISSUE TITLE
Declare List<Tuple> in View Error

### ISSUE THREAD

**Issue by RandyBuchholz** [aspnet/Mvc#5709]
I am unable to declare a List<T> of tuples in Views.

I tried a few variations of
```
@{
    var passed = new List<(string input, string output, string value)>{
	(input: "One", output: "Two", value: "three"),
	(input: "One", output: "Two", value: "three")
    };
}

and
 List<(string, string, string)> passed = new List<(string input, string output, string value)...

```
and I get a long list of errors at compile. 

![image](https://cloud.githubusercontent.com/assets/13967891/22234269/8530dae8-e1bc-11e6-91ce-459b5ffae159.png)

There is no problem if I pass and cast though. In my controller
```
ViewData["passed"] = new List<(string input, string output, string value)> {
	(input: "One", output: "Two", value: "three"),
	(input: "One", output: "Two", value: "three")
};
```
and then in my View
```
@{
    var passed = ViewData["Passed"] as List<ValueTuple<string, string, string>>;
}
```
everything works fine.

I get the same types of errors if I try in a function also
```
@functions{
	List<(string, string, string)> Passed() {
		return new List<(string input, string output, string value)>(){
		(input: "One", output: "Two", value: "three"), (input: "One", output: "Two", value: "three") };
	}
}
```

**Comment by pranavkm** [comment]
Mvc 1.1 and the .NET Core shared framework (`Microsoft.NETCore.App`) use an older build of Roslyn that supports at most C# 6. If you're targeting desktop (`net4x`), you could try referencing a newer version of [`Microsoft.CodeAnalysis.CSharp`](https://www.nuget.org/packages/Microsoft.CodeAnalysis.CSharp/2.0.0-rc2) in your project, although I'm not sure if you'd run into other issues at runtime.

@Eilon, we'll need to update the version of Roslyn in the the 1.2.0 shared framework if we'd like to support this. Are you aware of any ongoing discussions for this?

**Comment by Eilon** [comment]
@pranavkm not aware of any discussions, though I would imagine that is the plan. I'll start an email thread on this.

**Comment by Eilon** [comment]
Ok I heard back from the C# experts and it seems that the plan is to have C# 7.1 available in the "2.0" wave, which is what we're working on. That will include support for tuples.

(Other C# language feature status can be found here: https://github.com/dotnet/roslyn/blob/master/docs/Language%20Feature%20Status.md)

**Comment by Eilon** [comment]
Closing this issue as external because ultimately it's https://github.com/dotnet/corefx/ that needs to be updated w/ the new version of C#. Nothing we can do here but wait.
